src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/BuildAssetsUtils.cs

```diff
@@ -493,6 +495,31 @@ public static string GetMSBuildFilePath(PackageSpec project, RestoreRequest requ
 
                 props.AddRange(GenerateGroupsWithConditions(buildPropsGroup, isMultiTargeting, frameworkConditions));
 
+                // Create an empty PropertyGroup for package properties
+                var packagePathsPropertyGroup = MSBuildRestoreItemGroup.Create("PropertyGroup", Enumerable.Empty<XElement>(), 1000, isMultiTargeting ? frameworkConditions : Enumerable.Empty<string>());
+
+                var projectGraph = targetGraph.Graphs.FirstOrDefault();
+
+                // Distinct union of tool packages and packages with GeneratePathProperty=true
+                var packageIdsToCreatePropertiesFor = packagesWithTools.Union(projectGraph.Item.Data.Dependencies.Where(i => i.GeneratePathProperty).Select(i => i.Name)).Distinct(StringComparer.OrdinalIgnoreCase);
+
+                var localPackages = sortedPackages.Select(e => e.Value);
+
+                // Find the packages with matching IDs in the list of sorted packages, filtering out ones that there was no match for or that don't exist
+                var packagePathProperties = localPackages
+                    .Where(pkg => pkg?.Value?.Package != null && packageIdsToCreatePropertiesFor.Contains(pkg.Value.Package.Id) && pkg.Exists())
```

Comment:
I hit a case-sensitivity issue here.  I believe this contains call is the cause since I don't see how it handles if the winner of the distinct call above had different case than the value in Package.Id.

Reply:
https://github.com/NuGet/Home/issues/7843

Reply:
I think I've hit this too but wasn't able to figure out the problem.  I think you're right that its casing.  I'll submit a PR